types/src/schemas/stoatConfigSchemaRendered.json

```diff
@@ -93,17 +93,24 @@
     "static_hosting_plugin_rendered_map": {
       "type": "object",
       "additionalProperties": {
-        "$ref": "#/definitions/static_hosting_plugin_rendered"
+        "oneOf": [
+          { "$ref": "#/definitions/static_hosting_plugin_rendered" },
+          { "$ref": "#/definitions/static_hosting_plugin_rendered_variants" }
+        ]
       }
     },
     "static_hosting_plugin_rendered": {
       "type": "object",
-      "required": ["path", "link", "sha", "status"],
+      "required": ["with_variants", "path", "link", "sha", "status"],
       "properties": {
         "metadata": {
           "type": "object",
           "additionalProperties": true
         },
+        "with_variants": {
+          "type": "boolean",
+          "const": false
+        },
```

Comment:
Just so I understand, `with_variants` is used to distinguish between the. types for JSON schema handling since it can't tell them apart in the `oneOf` due to `additionalProperties`? 

Maybe we should use an enum to differentiate with a `type` in case there are more cases in the future?

Reply:
> Just so I understand, `with_variants` is used to distinguish between the. types for JSON schema handling since it can't tell them apart in the `oneOf` due to `additionalProperties`?

Yes. It's usually good practice to have an explicit field to distinguish between the `oneOf` types. Otherwise we may need to do complicated field check to figure out the schema in Typescript.

> Maybe we should use an enum to differentiate with a `type` in case there are more cases in the future?

Good point. That's what I did original, but I used a boolean because it's hard to choose a field name LOL. Will change to an const string.

Reply:
`type` with `DEFAULT` + `VARIANTS` maybe?